File: templates/hytale/optional/java/src/main/java/{{PACKAGE_PATH}}/{{MAIN_CLASS_NAME}}.java
package {{PACKAGE_NAME}};

import com.hypixel.hytale.logger.HytaleLogger;
import com.hypixel.hytale.server.core.plugin.JavaPlugin;
import com.hypixel.hytale.server.core.plugin.JavaPluginInit;
import com.hypixel.hytale.server.core.util.Config;

import java.util.logging.Level;

public class {{MAIN_CLASS_NAME}} extends JavaPlugin {

    private static final HytaleLogger LOGGER =
            HytaleLogger.forEnclosingClass();

    private final Config<ServerConfig> config;

    public {{MAIN_CLASS_NAME}}(JavaPluginInit init) {
        super(init);

        this.config = this.withConfig(
                "config",
                ServerConfig.CODEC
        );
    }

    @Override
    protected void setup() {
        LOGGER.at(Level.INFO)
                .log("[" + config.get().getModName() + "] Setting up...");

        LOGGER.at(Level.INFO)
                .log("[" + config.get().getModName() + "] Setup complete!");
    }

    @Override
    protected void start() {
        LOGGER.at(Level.INFO)
                .log("[" + config.get().getModName() + "] Started!");
    }

    @Override
    protected void shutdown() {
        LOGGER.at(Level.INFO)
                .log("[" + config.get().getModName() + "] Shutting down...");
    }
}

File: templates/hytale/optional/kotlin/src/main/kotlin/{{PACKAGE_PATH}}/{{MAIN_CLASS_NAME}}.kt
package {{PACKAGE_NAME}}

import com.hypixel.hytale.logger.HytaleLogger
import com.hypixel.hytale.server.core.plugin.JavaPlugin
import com.hypixel.hytale.server.core.plugin.JavaPluginInit
import com.hypixel.hytale.server.core.util.Config
import java.util.logging.Level

class {{MAIN_CLASS_NAME}}(
init: JavaPluginInit
) : JavaPlugin(init) {

    companion object {
        private val LOGGER =
            HytaleLogger.forEnclosingClass()
    }

    private val config: Config<ServerConfig> =
        withConfig(
            "config",
            ServerConfig.CODEC
        )

    override fun setup() {
        LOGGER.at(Level.INFO)
            .log("[${config.get().modName}] Setting up...")

        LOGGER.at(Level.INFO)
            .log("[${config.get().modName}] Setup complete!")
    }

    override fun start() {
        LOGGER.at(Level.INFO)
            .log("[${config.get().modName}] Started!")
    }

    override fun shutdown() {
        LOGGER.at(Level.INFO)
            .log("[${config.get().modName}] Shutting down...")
    }
}